samples/csharp_dotnetcore/40.timex-resolution/Resolution.cs

```diff
@@ -20,21 +20,12 @@ public static void Examples()
 
             Console.WriteLine(resolution.Values.Count);
 
-            Console.WriteLine(resolution.Values[0].Timex);
-            Console.WriteLine(resolution.Values[0].Type);
-            Console.WriteLine(resolution.Values[0].Value);
-
-            Console.WriteLine(resolution.Values[1].Timex);
-            Console.WriteLine(resolution.Values[1].Type);
-            Console.WriteLine(resolution.Values[1].Value);
-
-            Console.WriteLine(resolution.Values[2].Timex);
-            Console.WriteLine(resolution.Values[2].Type);
-            Console.WriteLine(resolution.Values[2].Value);
-
-            Console.WriteLine(resolution.Values[3].Timex);
-            Console.WriteLine(resolution.Values[3].Type);
-            Console.WriteLine(resolution.Values[3].Value);
+            foreach (var value in resolution.Values)
+            {
+                Console.WriteLine(value.Timex);
+                Console.WriteLine(value.Type);
+                Console.WriteLine(value.Value);
+            }
```

Comment:
The reason the 4 resolutions were explicitly called out in a fixed way is because this is a fixed size. So if you want to avoid code duplication the most correct factoring would be a for loop counting to 4 and not a foreach which implies 0 to n.

This resolution will result in 4 cases because there is ambiguity in the time part. This is mimicking LUIS which resolves dates backwards and forwards from now. So with ambiguity in time you get 4 resolutions otherwise you get 2.

Reply:
So change this something like:
```
for (var i=0; i<4; i++) { }
```
and it works. Otherwise I prefer the code that's there, it being more accurate.

Clearly this also needs more comments - I can add that.